### PR TITLE
fix MSVC/MinGW compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: mscore/palette/palettetree.cpp [hunk 1]
```diff
@@ -286,9 +286,9 @@ bool PalettePanel::read(XmlReader& e)
                   e.skipCurrentElement();
             else if (t == "type") {
                   bool ok;
-                  const int t = QMetaEnum::fromType<Type>().keyToValue(e.readElementText().toLatin1().constData(), &ok);
+                  const int t_ = QMetaEnum::fromType<Type>().keyToValue(e.readElementText().toLatin1().constData(), &ok);
```
Comment (anatoly-os) [inline mscore/palette/palettetree.cpp]: `type` or `localType` is better, IMHO. Using underline as an ending of a variable is a bad practice.

File: mscore/menus.cpp [hunk 2]
```diff
@@ -1187,8 +1187,8 @@ void MuseScore::showPalette(bool visible)
             updateIcons();
 
             paletteWidget = new PaletteWidget(getPaletteWorkspace(), getQmlUiEngine(), this);
-            QAction* a = getAction("toggle-palette");
-            connect(paletteWidget, &PaletteWidget::visibilityChanged, a, &QAction::setChecked);
+            QAction* a_ = getAction("toggle-palette");
```
Comment (anatoly-os) [inline mscore/menus.cpp]: `a_` equals to the `a` above. Both are obtained by calling `getAction("toggle-palette");` (see line 1173).

Comment (Jojo-Schmitz) [reply mscore/menus.cpp]: So dropping `QAction* ` here?

File: mscore/palette/palettetree.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: bit also longer. Global or outer types should have longer and more meaningful names, the more local/inner, the shorter they can get, as then their definition is within sight.
I could make it `t1` here or some such?

File: mscore/menus.cpp (hunk 2 above)
Comment (anatoly-os) [reply mscore/menus.cpp]: Try it and check whether something is broken :) Looks safe, but still.

File: mscore/palette/palettetree.cpp (hunk 1 above)
Comment (anatoly-os) [reply mscore/palette/palettetree.cpp]: I don't like numbers in variable names (as well as world C++ experts) :) It's better to rename the outer one.

Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: t1 is being used further up in the code, same file

Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: Hmm, here I guess `s` for the outer (being a string or rather a `QStringRef`) and `t`for the inner (being an int or `Type`) should be OK. Such a change is more invasive though, I try to keep those warning fixes as small as possible.

Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: Looking at some ofg my earlier PRs fixing warnings I found some more 'sins' of that sort.
Should I go back and fix those too?
One such case is in #4963, here we'd bettert change musescore.h `ScriptEngine* se     { 0 };`  to `ScriptEngine* scriptEngine     { 0 };` or even remove that apparently unused member alltogether

File: mscore/musescore.h [hunk 3]
```diff
@@ -366,7 +366,7 @@ class MuseScore : public QMainWindow, public MuseScoreCore {
 
       bool _midiinEnabled            { true };
       QList<QString> plugins;
-      ScriptEngine* se               { 0 };
+      ScriptEngine* scriptEngine     { 0 };
```
Comment (Jojo-Schmitz) [inline mscore/musescore.h]: Maybe we can even get rid of this entirrely, it doesn't seem to get used anywhere, @dmitrio95 ?

Comment (Jojo-Schmitz) [reply mscore/musescore.h]: OK, fine by me

File: libmscore/harmony.cpp [hunk 4]
```diff
@@ -644,8 +644,7 @@ const ChordDescription* Harmony::parseHarmony(const QString& ss, int* root, int*
       if (_leftParen || _rightParen)
             s = s.simplified();     // in case of spaces inside parentheses
 
-      int n = s.size();
-      if (n < 1)
+      if (s.size() < 1)
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply libmscore/harmony.cpp]: true

File: libmscore/harmony.cpp [hunk 5]
```diff
@@ -1677,11 +1676,10 @@ QString Harmony::userName() const
                   return QObject::tr("Roman numeral");
             case HarmonyType::NASHVILLE:
                   return QObject::tr("Nashville number");
-            case HarmonyType::STANDARD:
+            case HarmonyType::STANDARD: // fall-thru
             default:
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply libmscore/harmony.cpp]: yes, indeed better

File: mscore/palette/palettetree.cpp [hunk 6]
```diff
@@ -155,27 +155,27 @@ bool PaletteCell::read(XmlReader& e)
       visible = e.hasAttribute("visible") ? e.intAttribute("visible") : true;
 
       while (e.readNextStartElement()) {
-            const QStringRef& t1(e.name());
-            if (t1 == "staff")
+            const QStringRef& s(e.name());
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: OK

File: mscore/palette/palettetree.cpp [hunk 7]
```diff
@@ -270,34 +270,34 @@ bool PalettePanel::read(XmlReader& e)
       _name = e.attribute("name");
       _type = Type::Unknown;
       while (e.readNextStartElement()) {
-            const QStringRef t(e.name());
-            if (t == "gridWidth")
+            const QStringRef s(e.name());
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: yes

File: libmscore/harmony.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply libmscore/harmony.cpp]: But then we get a different warning: `control reached end of non-void function`, at least in MinGW

File: mscore/palette/palettetree.cpp (hunk 7 above)
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: But that gives another warning, C4458 in MSVC

**Comment by Jojo-Schmitz** [comment]
Adding fixed for a couple more and new warnings, adding an accessibility string too

**Comment by Jojo-Schmitz** [comment]
and a few new warnings just cropped up :-(

**Pull request merged**